File: binding/src/libqbasictimer.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qbasictimer.html
pub const qbasictimer = struct {

    /// New constructs a new QBasicTimer object.
    pub fn New() QtC.QBasicTimer {
        return qtc.QBasicTimer_new();
    }


    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QBasicTimer_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn IsActive(self: ?*anyopaque, ) bool {
        return qtc.QBasicTimer_IsActive(@ptrCast(self));
    }

    pub fn TimerId(self: ?*anyopaque, ) i32 {
        return qtc.QBasicTimer_TimerId(@ptrCast(self));
    }

    pub fn Start(self: ?*anyopaque, msec: i32, obj: ?*anyopaque) void {
        qtc.QBasicTimer_Start(@ptrCast(self), msec, @ptrCast(obj));
    }

    pub fn Start2(self: ?*anyopaque, msec: i32, timerType: i32, obj: ?*anyopaque) void {
        qtc.QBasicTimer_Start2(@ptrCast(self), msec, @intCast(timerType), @ptrCast(obj));
    }

    pub fn Stop(self: ?*anyopaque, ) void {
        qtc.QBasicTimer_Stop(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QBasicTimer_Delete(@ptrCast(self));
    }
};

File: binding/src/libqbasictimer.cpp
#include <QBasicTimer>
#include <QObject>
#include <qbasictimer.h>
#include "libqbasictimer.h"
#include "libqbasictimer.hxx"

QBasicTimer* QBasicTimer_new() {
	 return new QBasicTimer();
}

void QBasicTimer_Swap(QBasicTimer* self, QBasicTimer* other) {
	self->swap(*other);
}

bool QBasicTimer_IsActive(const QBasicTimer* self) {
	return self->isActive();
}

int QBasicTimer_TimerId(const QBasicTimer* self) {
	return self->timerId();
}

void QBasicTimer_Start(QBasicTimer* self, int msec, QObject* obj) {
	self->start(msec, obj);
}

void QBasicTimer_Start2(QBasicTimer* self, int msec, int timerType, QObject* obj) {
	self->start(msec, static_cast<Qt::TimerType>(timerType), obj);
}

void QBasicTimer_Stop(QBasicTimer* self) {
	self->stop();
}

void QBasicTimer_Delete(QBasicTimer* self) {
    delete self;
}

